resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -261,7 +281,8 @@ private[spark] object Client {
       .getOrElse(Array.empty[String])
     val appName = sparkConf.getOption("spark.app.name")
       .getOrElse("spark")
-    val kubernetesAppId = s"$appName-$launchTime".toLowerCase.replaceAll("\\.", "-")
+    val kubernetesAppId = sparkConf.getOption("spark.kubernetes.driver.pod.name")
```

Comment:
I think this is out of date with branch-2.1-kubernetes - since #331 has merged, the semantics of these variables have changed.